### PR TITLE
fix: ViewController swizzling before iOS 13

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

If an app targets, for example, iOS 13 or lower, the UIKit inits the initial/root view controller before the SentrySDK is initialized. Therefore, we manually call swizzle on the root view controller and all its descendant view controllers not to lose auto-generated transactions for the initial view controller.

I added UI tests to make sure the sample projects, iOS-SwiftUI, iOS-Swift, iOS-ObjectiveC, and tvOS-Swift launch without crashing.

## :bulb: Motivation and Context

Fixes GH-1132

## :green_heart: How did you test it?
Unit tests and tested with all sample projects except watchOS.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

**Review by brustolin — COMMENTED** [review]
I like the ideia, but doing some tests a found a problem.
If the root controller is an UINavigationViewControllers or UITabBarViewController (and probably UISplitViewController, i didn't try this), we would get the wrong controller to swizzle, and in the case of a UITabBarViewController, all tab view controllers are initialized together before the SDK.

We could check for this kind of root view controllers and act accordingly, at least those are the most used view controller. 

I'm wondering if there's some kind of user defined nested view controllers that would be a problem.

**Comment by philipphofmann** [comment]
> I like the ideia, but doing some tests a found a problem.
> If the root controller is an UINavigationViewControllers or UITabBarViewController (and probably UISplitViewController, i didn't try this), we would get the wrong controller to swizzle, and in the case of a UITabBarViewController, all tab view controllers are initialized together before the SDK.
> 
> We could check for this kind of root view controllers and act accordingly, at least those are the most used view controller.
> 
> I'm wondering if there's some kind of user defined nested view controllers that would be a problem.

@brustolin, but we only swizzle classes from the user app module. Shouldn't this do the trick?
https://github.com/getsentry/sentry-cocoa/blob/8bb1a674cf360f36b594ff15a34e640f14a613d9/Sources/Sentry/SentryUIViewControllerSwizziling.m#L60-L65

**Comment by brustolin** [comment]
> @brustolin, but we only swizzle classes from the user app module. Shouldn't this do the trick?

Because of this restriction we will not swizzle UINavigationViewController which is good, but because the code only try to swizzle the window rootViewController, the first controller to be used as the navigation controller child will not be swizzled, same for the controllers used as tabs in a UITabViewController.

**Comment by philipphofmann** [comment]
> > @brustolin, but we only swizzle classes from the user app module. Shouldn't this do the trick?
> 
> Because of this restriction we will not swizzle UINavigationViewController which is good, but because the code only tries to swizzle the window rootViewController, the first controller to be used as the navigation controller child will not be swizzled, same for the controllers used as tabs in a UITabViewController.

Do you mean we have to look for child view controllers of the rootViewController?

File: Sources/Sentry/SentryUIViewControllerSwizziling.m [hunk 1]
```diff
@@ -43,6 +44,75 @@ + (void)swizzleViewControllerInits
...
+                         andLevel:kSentryLevelDebug];
+        return;
+    }
+
+    NSArray<UIViewController *> *allViewControllers = rootViewController.descendantViewControllers;
+
+    for (UIViewController *viewController in allViewControllers) {
+        Class viewControllerClass = [viewController class];
+        if (viewControllerClass != nil) {
+            [SentryLog logWithMessage:@"Calling swizzleRootViewController."
```
Comment (brustolin) [inline Sources/Sentry/SentryUIViewControllerSwizziling.m:111]: l: I think it would be nice to have the name of the class here in the log being swizzled.

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**